Developer-Guide.md

```diff
@@ -0,0 +1,193 @@
+* [Warning](#warning)
+* [Assumptions](#assumptions)
+* [Build and install a Kata Containers runtime](#build-and-install-a-kata-containers-runtime)
+    * [Check hardware requirements](#check-hardware-requirements)
+    * [Enable full debug](#enable-full-debug)
+* [Build and install Kata proxy](#build-and-install-kata-proxy)
+* [Build and install Kata shim](#build-and-install-kata-shim)
+* [Create an image](#create-an-image)
+    * [Build a custom Kata agent - OPTIONAL](#build-a-custom-kata-agent---optional)
+    * [Get the osbuilder](#get-the-osbuilder)
+    * [Create a rootfs image](#create-a-rootfs-image)
+        * [Add a custom agent to the image - OPTIONAL](#add-a-custom-agent-to-the-image---optional)
+    * [Build the image](#build-the-image)
+* [Install the image](#install-the-image)
+* [Install guest kernel images](#install-guest-kernel-images)
+* [Update Docker configuration](#update-docker-configuration)
+* [Create a Kata Container](#create-a-kata-container)
+* [Problem determination](#problem-determination)
+* [Appendices](#appendices)
+    * [Checking Docker default runtime](#checking-docker-default-runtime)
+
+# Warning
+
+This document is written **specifically for developers**.
+
+# Assumptions
+
+- You are working on a non-critical test or development system.
+- You already have the following installed:
+  - [Docker](https://www.docker.com/).
+  - [golang](https://golang.org/dl) version 1.8.3 or newer.
+  - `make`.
+  - `gcc` (required for building the shim and runtime).
+
+- You have installed the Clear Containers `linux-container` and `qemu-lite`
```

Comment:
This part is a bit more error prone since we're referring to another install guide, but only a segment of it.  This is definitely a short term problem (ie, until we have packages available for these for Kata project), but I think it may be worth explicitly adding these in the meantime.  WDYT, @jodh-intel ?

Reply:
Yep - we really need https://github.com/kata-containers/packaging/issues/1 to be resolved.

I understand the general view but I'm not clear what you want me to add?

Reply:
I'd consider adding the steps for installing  qemu-lite and linux-containers.

Reply:
I actually left it purposely slightly vague since otherwise we need to be fully prescriptive (`apt-get`, `dnf`, `yum`, etc).

Once we have kata packages we can then start putting together install guides for the different distros as we did for [Clear Containers](https://github.com/clearcontainers/runtime/tree/master/docs).

FWIW, the same applies to installing docker, golang, etc.

Reply:
Hi @egernst - How about we keep this as-is for now since:

- it's what we already have in the wiki.
- it's temporary: we'll replace the vagueness with pointers to install guides once https://github.com/kata-containers/packaging/issues/1 is resolved?

Reply:
Works for me.